sway-core/src/sway.pest

```diff
@@ -157,7 +156,7 @@ str_type           =  { "str" ~ "[" ~ basic_integer ~ "]" }
 trait_bounds       =  {"where" ~ (generic_type_param ~ ":" ~ trait_name) ~ ("," ~ generic_type_param ~ ":" ~ trait_name)*}
 generic_type_param =  {ident}
 // Array size can be any constant u64 expression, but we don't properly support constant expressions.  See `array_elems rule above.
-array_type         =  {"[" ~ type_name ~ ";" ~ u64_integer~ "]"}
+array_type         =  {"[" ~ type_name ~ ";" ~ basic_integer ~ "u64"? ~ "]"}
```

Comment:
Hm, this doesn't seem correct. I believe we just wanted a number _without_ `u64` at all. `[bool; 5u64]` doesn't actually make any sense.
```suggestion
array_type         =  {"[" ~ type_name ~ ";" ~ basic_integer ~ "]"}
```

Reply:
You're right. The reason why I put the `"u64"?` here is because previously, `u64_integer` used to be `basic_integer ~ "u64"?` and I didn't want to change the old behavior.

Reply:
Fixed.